runtime/interpreter/decode.go

```diff
@@ -1044,21 +1210,18 @@ func (d *DecoderV4) decodeCapability(v interface{}) (CapabilityValue, error) {
 		return CapabilityValue{}, fmt.Errorf("invalid capability path: %T", path)
 	}
 
+	// Decode borrow type at array index encodedCapabilityValueBorrowTypeFieldKey
+
 	// borrow type (optional, for backwards compatibility)
 	// Capabilities used to be untyped, i.e. they didn't have a borrow type.
 	// Later an optional type paramater, the borrow type, was added to it,
 	// which specifies as what type the capability should be borrowed.
 	//
 	// The decoding must be backwards-compatible and support both capability values
 	// with a borrow type and ones without
-	var borrowType StaticType
-
-	if field3 := encoded[encodedCapabilityValueBorrowTypeFieldKey]; field3 != nil {
-
-		borrowType, err = d.decodeStaticType(field3)
-		if err != nil {
-			return CapabilityValue{}, fmt.Errorf("invalid capability borrow type encoding: %w", err)
-		}
+	borrowType, err := d.decodeStaticType()
+	if err != nil {
+		return CapabilityValue{}, fmt.Errorf("invalid capability borrow type encoding: %w", err)
 	}
```

Comment:
Is the borrow type here still optional after the change?

Reply:
Yes.  After reading your code review, I removed optional static type decoding from decodeStaticType().  That is now done in 3 places which need optional static type decoding.